packages/preferences-persistence/package.json

```diff
@@ -28,7 +28,8 @@
 	"sideEffects": false,
 	"dependencies": {
 		"@babel/runtime": "^7.16.0",
-		"@wordpress/api-fetch": "file:../api-fetch"
+		"@wordpress/api-fetch": "file:../api-fetch",
+		"@wordpress/compose": "file:../compose"
```

Comment:
This is where I'm a little bit concerned that `debounce` is part of the `@wordpress/compose` package. As suddenly, this package also depends on `@wordpress/element`, and therefore `react` and `react-dom`. It isn't an issue outside of WordPress as build tools will only bundle `debounce`, but inside WP it makes it more difficult to use `debounce` in lower-level packages.

Reply:
Interesting. On the other hand, won't this happen for other modules from packages if we theoretically want to include them in lower-level packages? There's always the possibility of that, but are we really going to use `debounce` in `@wordpress/element`?

In any case, I'm flexible - what would you recommend instead? Yet another separate package outside of the repository?

Reply:
The same issue applies to `compose` from `@wordpress/compose`. Ideally, those utils would be a part of the standard JavaScript library so every project doesn't need to implement them 😄

I hope it's what eventually will happen as we see it with the removal of `lodash`.

Reply:
Having first-party support for `compose()` and `debounce()` in JavaScript is ideal, of course. I guess for now, a better solution with less dead weight would be a third-party package. I'll think about that.

Reply:
I guess it's fine to start by including `debounce` in `@wordpress/compose` as we did with `compose` in the past. The reasoning with `useDebounce` + `debounce` convinced me.